zig/shell.nix

```diff
@@ -0,0 +1,67 @@
+{ pkgs ? import <nixpkgs> {} } :
+let
+	# Right. Getting this to work took more effort than all the other shells put together.
+	# Things that didn't work:
+	# - Directly using the Zig package, pinned to any version. Nix is "known as the most up to date distribution", but this project uses a *nightly* build of Zig, so that's not there.
+	# - Building Zig from source. The latest version of Zig itself uses LLVM+Clang 15, which is brand new and also not yet in Nix repos. There was a brief window in late August 2022 where Zig could be built with LLVM 14, but the most recent commit that works with that doesn't work with this project.
```

Comment:
ah, sorry; still working on [this PR](https://github.com/NixOS/nixpkgs/pull/194634) :stuck_out_tongue: 

---

Regardless I think `zig-overlay` is definitely the best option for this project.